Add initial query page

Arif-Khalid opened the pull request:
### Changes made
* Added query page containing placeholder app logo
* Used controlled input behaviour on query input
* Added prettier package and put prettier config in .prettierrc file
* Add format command to package.json
* Add simple grid styling to content
* Add temporary fake API call returning fake data when query sends

###
Before sending query
<img src=https://github.com/nknguyenhc/luminous-zephyr/assets/88131400/0312bae9-d0c7-4829-a594-44c286f8272d height=500/>
After sending query
<img src=https://github.com/nknguyenhc/luminous-zephyr/assets/88131400/46eb5735-94ef-4263-82ec-5f1e3c720171 height=500/>

nknguyenhc reviewed (commented):
@Arif-Khalid Is it possible for you to add the `.prettierrc` file, and possibly an `npm` command to do style correction?

Arif-Khalid commented:
> Also, I believe file naming convention should be spinal case, e.g. `content-component`, `query-component`.

Renamed appropriately

> @Arif-Khalid Is it possible for you to add the `.prettierrc` file, and possibly an `npm` command to do style correction?

Added format command to package.json and moved prettier config to .prettierrc

Arif-Khalid commented:
After the most recent review I have also added simple grid styling, a button that sends the query and a custom hook that calls a simulated fetch.

nknguyenhc commented on frontend/src/hooks/useProducts.ts:
```diff
@@ -0,0 +1,94 @@
+export function useProducts() {
+  const [products, setProducts] = useState([] as Product[])
+
+  function sendQuery(query: string) {
+    if (!query) {
+      //TODO add error handling
+      return
+    }
+
+    // Fake API call
+    setTimeout(() => {
+      setProducts(testProducts)
+      alert(`Query Received: ${query}`)
+    }, 1000)
+  }
```
Better to make nested functions `useCallback` hooks. The `useCallback` hooks allow the function definition to cached between renders, while function declaration refreshes the function declaration on every re-render.

Arif-Khalid replied:
Good call I have refactored to use useCallback

nknguyenhc commented on frontend/src/components/query-component/query-component.tsx:
```diff
@@ -0,0 +1,34 @@
+        disableUnderline={true}
+      ></Input>
+      <IconButton
+        onClick={() => {
+          sendQuery(query)
+          setQuery('')
+        }}
```
Should abstract out the callbacks as hooks. Definition of a listener declared with a lambda expression is never refreshed, while definition of a listener with a hook is refreshed when necessary.

```ts
export function QueryComponent({ sendQuery }: QueryComponentProps) {
  const handleClick = useCallback(() => {
    sendQuery();
    setQuery('');
  }, [sendQuery]);

  // Other parts

  return (
    // ...
    <IconButton
      onClick={handleClick}
      className="query-send-icon-button"
      aria-label="send"
    />
    // ...
  );
}
```

Arif-Khalid replied:
I agree I have abstracted to a memoized function appropriately

nknguyenhc reviewed (commented):
A few comments on the UI:

![image](https://github.com/nknguyenhc/luminous-zephyr/assets/87511888/a0664f27-7d2c-4b0f-b028-23ec1364cfec)

1. The screen before searching should not have any scroll bar. This can be done by setting the `overflow` property to `overflow: auto` instead of `overflow: scroll`.
2. User should be able to press enter to do the search, instead of having to press the send button. I would suggest wrapping the input field in a `form` element, because the `form` element handles hitting enter for you. But up to you how you want to resolve.

Besides, some comment on quality of code below.

Arif-Khalid commented:
> A few comments on the UI:
> 
> ![image](https://private-user-images.githubusercontent.com/87511888/341116212-a0664f27-7d2c-4b0f-b028-23ec1364cfec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4MTE1MjEsIm5iZiI6MTcxODgxMTIyMSwicGF0aCI6Ii84NzUxMTg4OC8zNDExMTYyMTItYTA2NjRmMjctN2QyYy00YjBmLWIwMjgtMjNlYzEzNjRjZmVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDE1MzM0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlYjE4ZjQzYzBjZDUyNmY4YjVlYTIxYWZkNmVkM2M2NWFhYzk0NjRhOGFkMzFiYjVhZDczYThiZjlhM2I0YzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.P5jTradkpHs1bdEzJuuW3V10HODAw5A_2hK6Rm9wvZo)
> 
> 1. The screen before searching should not have any scroll bar. This can be done by setting the `overflow` property to `overflow: auto` instead of `overflow: scroll`.
> 2. User should be able to press enter to do the search, instead of having to press the send button. I would suggest wrapping the input field in a `form` element, because the `form` element handles hitting enter for you. But up to you how you want to resolve.
> 
> Besides, some comment on quality of code below.

Good observations which have been addressed by my latest commits

nknguyenhc reviewed (approved):
LGTM!
